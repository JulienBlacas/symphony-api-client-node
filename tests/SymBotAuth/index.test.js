const SymBotAuth = require('../../lib/SymBotAuth');

jest.mock('fs');
jest.mock('https');
const mockFs = require('fs');
const mockHttps = require('https');

describe('SymBotAuth', () => {
  beforeEach(() => {
    mockFs.readFileSync = jest.fn((path, encoding) => {
      return "Contents of " + path;
    });

    debugger;
    mockHttps.request = jest.fn(() => {
      return {
        end: jest.fn()
      };
    });
  });

  afterEach(() => {
    jest.resetModules();
  });

  describe('sessionAuthenticate', () => {
    describe('when the auth type is not rsa', () => {
      test.only('POSTs to /sessionauth/v1/authenticate with the correct parameters', () => {
        SymBotAuth.sessionAuthenticate({
          botCertPath: "/",
          botCertName: "I am the cert",
          botKeyName: "I am the key",
          sessionAuthHost: "https://session.example.com",
          sessionAuthPort: 1234,
          botCertPassword: "Passphrase",
          sessionAuthProxy: "agent"
        });

        expect(mockHttps.request.mock.calls).toMatchSnapshot();
      });
    });
  });

  describe('kmAuthenticate', () => {
    describe('when the auth type is not rsa', () => {
      test.only('POSTs to /keyauth/v1/authenticate with the correct parameters', () => {
        SymBotAuth.kmAuthenticate({
          botCertPath: "/",
          botCertName: "I am the cert",
          botKeyName: "I am the key",
          keyAuthHost: "https://key.example.com",
          keyAuthPort: 5678,
          botCertPassword: "Passphrase",
          keyAuthProxy: "agent"
        });

        expect(mockHttps.request.mock.calls).toMatchSnapshot();
      });
    });
  });
});
