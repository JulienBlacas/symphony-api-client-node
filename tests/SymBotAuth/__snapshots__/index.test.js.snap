// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SymBotAuth kmAuthenticate when the auth type is not rsa POSTs to /keyauth/v1/authenticate with the correct parameters 1`] = `
Array [
  Array [
    Object {
      "agent": "agent",
      "cert": "Contents of /I am the cert",
      "hostname": "https://key.example.com",
      "key": "Contents of /I am the key",
      "method": "POST",
      "passphrase": "Passphrase",
      "path": "/keyauth/v1/authenticate",
      "port": 5678,
    },
    [Function],
  ],
]
`;

exports[`SymBotAuth sessionAuthenticate when the auth type is not rsa POSTs to /sessionauth/v1/authenticate with the correct parameters 1`] = `
Array [
  Array [
    Object {
      "agent": "agent",
      "cert": "Contents of /I am the cert",
      "hostname": "https://session.example.com",
      "key": "Contents of /I am the key",
      "method": "POST",
      "passphrase": "Passphrase",
      "path": "/sessionauth/v1/authenticate",
      "port": 1234,
    },
    [Function],
  ],
]
`;
